[26기 정지후] Add: Westagram login page 구현

j-jhoo opened the pull request:
- 로그인 페이지 레이아웃 작성 완료
- 로그인 페이지 스타일 속성 적용 완료
- 로그인 페이지 유효성 검사 기능 관련 로직 작성 완료
- 로그인 페이지 아이디 비밀번호 입력시 로그인 버튼 활성화 기능 완료

- 메인 페이지 레이아웃 작성 완료
- 메인 페이지 스타일 속성 적용 완료
- 메인 페이지 리팩토링 완료
- 메인 페이지 댓글 엔터키, 게시버튼 누를 경우 입력되게 기능 활성화 완료

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 페이지 리팩토링 완료
- 메인페이지 js 엔터키 누를경우 댓글 입력 기능 구현 완료
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 메인 페이지 js 엔터키 눌러서 댓글 입력시 댓글이 두줄씩 작성되길래 기존에 있던 게시버튼 입력시 댓글 구현되는 함수를 지우고 엔터키 함수이 추가로 넣어서 구현 해본 결과 정상적으로 수정 완료 되었다. 
- 
☆## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x ] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

dipsiiiiiiiiii commented on students/13th/leedonghun/index.html:
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+    <title>Westagram by pdl39</title>
+    <link href="https://fonts.googleapis.com/css2?family=Arimo:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
+    <link href="style.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <script src="script.js"></script>
+  </body>
+</html>
```
index.html 파일이 필요한가요!?

pdl39 replied:
아 네 삭제하겠습니다!!

pdl39 replied:
e6f08ddc23ee0613df15b7c37b4587ffe83c8c70

dipsiiiiiiiiii commented on students/13th/leedonghun/js/login.js:
```diff
@@ -0,0 +1,48 @@
+
+function activateLoginButton() {
+  const loginButton = document.getElementById('login-button');
```
Id로 다 사용해주셨는데 id는 혼자 개발할 때는 상관없지만 여러 명이 동시에 개발하면 id가 겹칠 수 있기 때문에
사용하지 않는 걸 권장합니다.
클래스로 다 바꿔주세요 😄

pdl39 replied:
네!

pdl39 replied:
5d04a399d1d6c15a302c281bac540fca885ab41c

dipsiiiiiiiiii commented on students/13th/leedonghun/js/login.js:
```diff
@@ -0,0 +1,48 @@
+  }
+}
+
+const idBox = document.getElementById('username-box');
+const pwBox = document.getElementById('pw-box');
```
idBox를 바로 위의 함수에서 사용하고 있어서 이런 정의는 제일 상단에 적어주세요!

pdl39 replied:
네!!

pdl39 replied:
76c78c4d8df8ac5eedd9cb45e9092b807983a1f7

dipsiiiiiiiiii commented on students/13th/leedonghun/js/login.js:
```diff
@@ -0,0 +1,48 @@
+
+function activateLoginButton() {
+  const loginButton = document.getElementById('login-button');
+  if (idBox.value.length > 0 && pwBox.value.length > 5) {
+    loginButton.style.backgroundColor = '#0095f6';
+    loginButton.style.cursor = 'pointer';
+    loginButton.disabled = false;
+  } else {
+    loginButton.style.backgroundColor = '#b2dffc';
+    loginButton.disabled = true;
+  }
```
너무 잘 적어주셨네요!
다르게 적어볼 수 있는 방법도 있습니다!

idBox.value.length > 0 && pwBox.value.length > 5 조건을 하나의 변수로 담으면 boolean 값이 되겠죠?
이거와 삼항 연산자를 활용하면 더 깔끔하게 적을 수도 있습니다!
물론 지금도 너무 깔끔하게 잘 적어주셨어요!
```suggestion
const isValid = idBox.value.length > 0 && pwBox.value.length > 5;
login.button.style.backgroundColor = isValid ? '#0095f6' : '#b2dffc';
loginButton.style.cursor = isValid ? 'pointer' : "";
loginButton.disabled = !isValid;
```

pdl39 replied:
와 이게 이렇게 깔끔하게 정리 가능하군요!! ㅋㅋ 감사합니다!

pdl39 replied:
위 코드는 참고용으로 이대로 놔두고 아래 또 다른 케이스는 알려주신대로 바꿔봤습니다~~
6684c99a6e8d9ec32ca9c66231a6b133f151c533

dipsiiiiiiiiii commented on students/13th/leedonghun/js/login.js:
```diff
@@ -0,0 +1,48 @@
+pwBox.addEventListener('input', activateLoginButton);
+
+/*
+function inputOutlineOnClick(e) {
```
쓰지 않는 코드는 다 지워주세요 :)

pdl39 replied:
네!!!

pdl39 replied:
76c78c4d8df8ac5eedd9cb45e9092b807983a1f7

dipsiiiiiiiiii commented on students/13th/leedonghun/js/login.js:
```diff
@@ -0,0 +1,48 @@
+const idBox = document.getElementById('username-box');
+const pwBox = document.getElementById('pw-box');
+
+idBox.addEventListener('input', activateLoginButton);
+pwBox.addEventListener('input', activateLoginButton);
```
이렇게 하셔도 되고 idBox와 pwBox를 감싸는 div 태그 하나 만들고
그 div에 이벤트를 한 번만 걸어주실 수도 있습니다!

pdl39 replied:
네 참고하겠습니다!

dipsiiiiiiiiii commented on students/13th/leedonghun/js/main.js:
```diff
@@ -0,0 +1,44 @@
+
+function activatePostButton(e) {
+  const postButton = document.querySelector('.post-button');
+  if (e.target.value.length > 0) {
+    postButton.style.opacity = 1;
+    postButton.style.cursor = 'pointer';
+  } else {
+    postButton.style.opacity = 0.3;
+    postButton.style.cursor = 'default';
+  }
```
여기도 너무 잘 해주셨는데 위의 리뷰 참고해서 한 번 고쳐보세요!

pdl39 replied:
네!
6684c99a6e8d9ec32ca9c66231a6b133f151c533

dipsiiiiiiiiii commented on students/13th/leedonghun/js/main.js:
```diff
@@ -0,0 +1,44 @@
+}
+
+const postCommentBox = document.querySelector('.post-comment-box');
+postCommentBox.addEventListener('input', activatePostButton);
```
input이라는 이벤트 이름이 있나요?

pdl39 replied:
<input>, <select>,  <textarea> 태그에 대해서 value값이 바뀔때 fire되는 input 이벤트가 있더라고요! keyup 말고도 쓸 수 있는 이벤트 있나 찾아보다가 구글링 해서 찾았습니다~ ㅎㅎ https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event

pdl39 replied:
\<input\>,\<select\> , \<textarea\> 태그에 대해서 value값이 바뀔때 fire되는 input 이벤트가 있더라고요! keyup 말고도 쓸 수 있는 이벤트 있나 찾아보다가 구글링 해서 찾았습니다~ ㅎㅎ 
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event

dipsiiiiiiiiii commented on students/13th/leedonghun/js/main.js:
```diff
@@ -0,0 +1,44 @@
+
+
+function changeHeartColor(e) {
+  console.log('clicked heart!');
```
테스트 끝난 콘솔은 지워주세요!
유저가 개발 과정을 보면 안되겠죠!?

pdl39 replied:
네!

pdl39 replied:
a6c3871e8395a085cd08e10af04ee9a530b35ee1

dipsiiiiiiiiii commented on students/13th/leedonghun/js/main.js:
```diff
@@ -0,0 +1,44 @@
+
+function changeHeartColor(e) {
+  console.log('clicked heart!');
+  function isLiked() {
+    if (e.target.src = "../img/main/nav-menu_heart_icon.png") {
+      return false;
+    } else if (e.target.src = "../img/main/onclick_icons/nav_menu_heart_icon_red.png") {
+      return true;
+    }
+  }
+  
+  if (!isLiked()) {
+    console.log('not yet liked!');
+    e.target.src = "../img/main/onclick_icons/nav_menu_heart_icon_red.png";
+  } else if (isLIked()) {
+    e.target.src = "../img/main/nav_menu_heart_icon.png";
+  }
```
이렇게 함수를 나눠서 하신 이유가 있을까요!?

pdl39 replied:
수정한 파일엔 여러 아이콘에 일괄적으로 적용할 수 있는 changeIconColor() 함수를 만들었는데 처음 changeHeartColor() 함수 만들때도 이렇게 일괄적으로 적용할 수 있는 함수를 생각을 하고 있어서 분리했습니다~

pdl39 replied:
isLiked()로 나눠서 changeHeartColor() 안에서 불러오면 단순하게 'img src가 뭐일때 뭐로 바꾼다' 로 돼있는것 보다 좀 더 직관적인 것 같애서 나눴습니다!

dipsiiiiiiiiii commented on students/13th/leedonghun/style/login.css:
```diff
@@ -0,0 +1,270 @@
+
+/* login.css styles */
+body {
+  margin: 0 20px;
```
body에 마진을 주시면 nav바 같은 레이아웃은 body 바깥으로 들어가야 해서 좋지 않습니다.
body는 화면을 채워주세요!

pdl39 replied:
네! 093518129de2ed1c9d0a87919851765e9ba0fb6f

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+  <body>
+    <section id="login-main">
+      <section id="login-phones">
+        <div class="phone-frame">
```
이 div가 없어도 될 것 같습니다!

pdl39 replied:
네 삭제했습니다! 1f5886dbe1eb42d1e4b7e2188fa0545fd1dcba38

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+    <section id="login-main">
+      <section id="login-phones">
+        <div class="phone-frame">
+          <img class="phone-frame-img" src="img/login/westagram_login_page_phones.png" alt="two phones overlapped showing preview of Instagram">
+          <div class="phone-screen"></div>
```
같은 이미지인데 하나는 img태그 사용하고 하나는 div태그에 백그라운드 이미지 주신 거 같은데
어떤 기준으로 하셨나요?

pdl39 replied:
phone-screen div는 css animation keyframe으로 이미지 변화 주기위해서 이것만 백그라운드 이미지로 주었습니다~

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+          <input type="password" id="pw-box" placeholder="Password">
+          <button id="login-button" disabled=true>Log In</button>
+          <div class="or-box">
+            <div class="or-line left"></div>
```
여기는 block 부분이 아니라서 span 사용하실 수도 있을 것 같습니다!

pdl39 replied:
네 수정했습니다~~ 3981887843c009d161c5570f07a3f00e09516f04

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+          <button id="login-button" disabled=true>Log In</button>
+          <div class="or-box">
+            <div class="or-line left"></div>
+            <div class="or"><p>OR</p></div>
+            <div class="or-line right"></div>
```
여기도 마찬가지!

pdl39 replied:
네! 3981887843c009d161c5570f07a3f00e09516f04

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+            <div class="or-line right"></div>
+          </div>
+          <div class="login-with-facebook">
+            <div class="fb-logo"><img src="img/login/facebook_logo.png" alt="facebook logo"></div>
```
여기도 꼭 div일 필요가 있을까요!?

pdl39 replied:
수정했습니다! e5c83ca174db264069c35f23165220982c160bc7

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+          </div>
+          <div class="login-with-facebook">
+            <div class="fb-logo"><img src="img/login/facebook_logo.png" alt="facebook logo"></div>
+            <div class="fb-text"><a href="#"><p>Log in with Facebook</p></a></div>
```
여기도 마찬가지! div가 들어가야만 하나요!?

pdl39 replied:
e5c83ca174db264069c35f23165220982c160bc7

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+            <a href="#"><p>Forgot password?</p></a>
+          </div>
+        </div>
+        <div class="signup-box">
+          <p class="no-account-question">Don't have an account? <span class="signup-text"><a href="#">Sign up</a></span></p>
+        </div>
+        <div class="get-the-app">
+          <p>Get the app.</p>
+        </div>
```
p 태그를 div태그로 감싸야 할 이유가 있을까요?🤔

pdl39 replied:
수정했습니다~ b97da061b5fc158a476e5dd0efdab3baec53e8f3

dipsiiiiiiiiii commented on students/13th/leedonghun/login.html:
```diff
@@ -0,0 +1,68 @@
+    </section>
+    <footer id="footer">
+      <ul class="footer-menus">
+        <a href="https://about.instagram.com" target="_blank"><li class="footer-menu">ABOUT</li></a>
+        <a href="https://help.instagram.com"><li class="footer-menu">HELP</li></a>
+        <a href="https://about.instagram.com/blog"><li class="footer-menu">PRESS</li></a>
+        <a href="https://www.instagram.com/developer/"><li class="footer-menu">API</li></a>
+        <a href="https://www.instagram.com/about/jobs/"><li class="footer-menu">JOBS</li></a>
+        <a href="https://help.instagram.com/519522125107875"><li class="footer-menu">PRIVACY</li></a>
+        <a href="https://help.instagram.com/581066165581870"><li class="footer-menu">TERMS</li></a>
+        <a href="https://www.instagram.com/explore/locations/"><li class="footer-menu">LOCATIONS</li></a>
```
ul 태그는 li 태그를 자식으로 가져서 li태그가 a 태그를 가지고 있게 바꿔주세요!

pdl39 replied:
수정했습니다~ c96f9fe59c3e3f061ac2f40e9caed38f69e99b80

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+  </head>
+  <body>
+    <!-- ************************* Nav Bar ************************* -->
+    <nav id="nav-bar">
```
nav 태그 좋습니다!

pdl39 replied:
😃

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+    <nav id="nav-bar">
+      <div id="nav-items-container">
+        <div id="nav-logo">
+          <img src="img/main/logo_text.png" alt="insta text logo" class="nav-logo">
```
img 태그 속성의 순서를 신경써주세요!
src 속성이 가장 중요해 보이지만 단순 경로이기 때문에 가장 마지막에 오는 게 좋습니다.
alt 속성이 오히려 이 이미지 태그를 설명해 주는 태그이기 때문에 alt 속성을 제일 앞으로 해주세요!

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+        </div>
+      </div>
+    </nav>
+
+
```
불필요한 빈 칸은 모두 지워주세요!

pdl39 replied:
네~ 845ddf4e396038c26249c35187fadbb054c33e93

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+
+    <!-- ************************* Main Section - Story/Feeds & Side ************************* -->
+    <section id="main">
+
```
여기 빈 칸도! 다른 곳은 전부 다 찾아서 지워주세요~

pdl39 replied:
네~ 845ddf4e396038c26249c35187fadbb054c33e93

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+          <img src="img/main/nav_search_icon.png" alt="search icon" class="nav-search-icon">
+        </div>
+        <div id="nav-menu">
+          <div class="nav-menu-item home">
+            <img src="img/main/nav_menu_home_icon.png" alt="nav bar home menu icon" class="menu-item-icon home-icon">
+          </div>
+          <div class="nav-menu-item dm">
+            <img src="img/main/nav_menu_dm_icon.png" alt="nav bar dm icon" class="menu-item-icon dm-icon">
+          </div>
+          <div class="nav-menu-item explore">
+            <img src="img/main/nav_menu_explore_icon.png" alt="nav bar explore menu icon" class="menu-item-icon explore-icon">
+          </div>
+          <div class="nav-menu-item heart">
+            <img src="img/main/nav_menu_heart_icon.png" alt="nav bar heart menu icon" class="menu-item-icon heart-icon">
+          </div>
+          <div class="nav-menu-item avatar">
+            <img src="img/main/account_profile_pic.png" alt="nav bar avatar profile picture" class="menu-item-icon avatar-icon">
+          </div>
```
각 img 태그를 전부 div로 감싸야 할까요!?

pdl39 replied:
img 마다 크기가 다 달라서 div 안에 넣어주고 div container 에 대해 크기(width %)랑 위치(display: flex, justify-content: center, align-items: center)를 지정해서 딱 원하는 크기/모양으로 보여주려고 각각 다 div 안에 넣었는데 혹시 div container를 따로 주지 않고 img 태그 자체로 크기/proportion을 미세하게 조정할 수 있는 방법이 있을까요??

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+      <!-- *************** Story & Feeds Section ************** -->
+      <section id="feeds-section">
+        <!-- ***** Story ***** -->
+        <div class="story">
+          <!-- One Story Element (reusable) >>> -->
+          <div class="story-element story-1">
```
스토리 리스트가 나오니까 ul, li 태그 사용하시는 건 어떨까요?

pdl39 replied:
네 수정했습니다! 9e620cfe7ff9dfbff1f6eb9bd653eddba223e13f

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+              </div>
+              <div class="feed-option"><img src="img/main/three_dots_horizontal.png" alt="feed option button" class="feed-option"></div>
+            </div>
+            <div class="feed-main">
+              <img src="img/main/feed_pics/pizza2.png" alt="feed picture" class="feed-main-pic">
+            </div>
```
img 태그만으로도 충분할 것 같습니다!

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+              <img src="img/main/feed_pics/pizza2.png" alt="feed picture" class="feed-main-pic">
+            </div>
+            <div class="feed-actions">
+              <div class="feed-action feed-like">
+                <img src="img/main/nav_menu_heart_icon.png" alt="feed like heart icon" class="feed-action-heart">
+              </div>
+              <div class="feed-action feed-comment">
+                <img src="img/main/nav_menu_comment.png" alt="see comment icon" class="feed-action-comment">
+              </div>
+              <div class="feed-action feed--dm">
+                <img src="img/main/nav_menu_dm_icon.png" alt="" class="feed-action-dm">
+              </div>
+              <div class="feed-action feed-tag">
+                <img src="img/main/nav_menu_tag_icon.png" alt="" class="feed-action-tag">
+              </div>
+            </div>
+            <div class="likes-count">
+              <p class="no-likes"># likes</p>
+            </div>
```
여기도 마찬가지!
div 태그로 일일이 다 감싸 줄 필요는 없을 것 같아요!

dipsiiiiiiiiii commented on students/13th/leedonghun/main.html:
```diff
@@ -0,0 +1,371 @@
+              <a href="" class="feed-profile-id">usrnm123</a>
+              <p class="feed-exp">pizza very yummy</p>
+            </div>
+            <div class="feed-comments">
+              <ul class="feed-comments">
```
여기도 ul만 있어도 괜찮을 것 같아요!

pdl39 replied:
네! 0de559ba7112fd7b670db89104d5e1da773fdb46

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다!
전체적으로 너무 깔끔하게 짜주셨어요!
다만 div 태그는 필요한 곳에서만 쓰실 수 있게 한 번 더 생각하시면서 짜주시면 더 좋을 것 같습니다!
리뷰 반영해주세요 :)

j-jhoo committed:
Add: [Mission 1] 로그인 페이지 완료
- 로그인 페이지 레이아웃 작성 완료
- 로그인 페이지 스타일 속성 적용 완료
- 로그인 페이지 유효성 검사 기능 관련 로직 작성 완료
- 로그인 페이지 아이디 비밀번호 입력시 로그인 버튼 활성화 기능 완료

j-jhoo committed:
main page css를 수정하고 리팩토링을 진행하였습니다.
js코드를 입력해서 댓글입력 후 게시버튼을 눌러서 입력까지 완료 하였습니다.
delete버튼이 생성되지 않는 문제는 완료하지 못했습니다.

j-jhoo committed:
test

j-jhoo committed:
main.js에서input에 값을 넣으면 게시 버튼이 활성화되기를 추가했다.

j-jhoo committed:
main.js에서 enter키를 눌렀을때도 댓글이 입력 될 수 있는 코드를 추가 했다.
엔터키를 눌렀을때 적용이 되었지만 두줄씩 입력이되서 고민하다가, 게시버튼을
누르면 입력되는 함수를 지우고 새로운 코드로 바꿔서 적용 시켰다.

j-jhoo committed:
"main js 수정"
댓글 엔터 입력시 한국어는 두줄이 입력되던 문제를 이벤트 'keyup'으로 변경하면서
수정 완료하였다.
